### DISCUSSION TITLE
Getting Error no attribute 'dataset_id' when using BigQueryToMsSqlOperator when there is no attribute dataset_id listed

### DISCUSSION THREAD

**Discussion by dudutch** [apache/airflow#19047] [Q&A]
I'm using the example dag example_bigquery_to_mssql.py 

My task is set up as follows:
```
    bigquery_to_mssql = BigQueryToMsSqlOperator(
        task_id="bigquery_to_mssql",
        gcp_conn_id="gcp_************",
        mssql_conn_id="******-db-dev",
        source_project_dataset_table=f'{DATASET_NAME}.{ORIGIN}',
        mssql_table=destination_table,
        replace=True,
)
```
I get the error
```
AttributeError: 'BigQueryToMsSqlOperator' object has no attribute 'dataset_id'
```
when trying to run the dag.  

There is a dataset_id in template_field but it does not seem to be needed in Parameters for teh BigQuere to MsSQL Operator.
https://airflow.apache.org/docs/apache-airflow-providers-google/6.0.0/_api/airflow/providers/google/cloud/transfers/bigquery_to_mssql/index.html

Has anyone else experienced this?

**Comment by josh-fell** [comment]
This looks like a bug in the `BigQueryToMsSqlOperator` presumably when attempting to render the template fields for the `TaskInstance` when the task executes.  The `table_id` and `dataset_id` items in the operator's `template_fields` class attribute should be removed (or even simply replaced by `source_project_dataset_table`) as they are not attributes of the operator.

As a potential workaround you should be able to subclass the `BigQueryToMsSqlOperator` to overwrite the `template_fields` class attribute so `dataset_id` and `table_id` are removed/replaced and use that new operator in your pipeline.

**Reply by josh-fell** [reply]
@dudutch I created #19052 to fix this bug. Thanks for reporting this!

**Reply by dudutch** [reply]
Thank you Josh!  I think there may be one other issue similar to what was addressed for mysql in https://github.com/apache/airflow/issues/17198

```
    def _bq_get_data(self):
        hook = BigQueryHook(
            bigquery_conn_id=self.gcp_conn_id,
            delegate_to=self.delegate_to,
            location=self.location,
            impersonation_chain=self.impersonation_chain,
        )
        table_ref = TableReference.from_string(self.source_project_dataset_table)
        self.log.info('Fetching Data from:')
        self.log.info('Project: %s, Dataset: %s, Table: %s',table_ref.project, table_ref.dataset_id, table_ref.table_id)

        i = 0
        while True:
            response = hook.list_rows(
                dataset_id=table_ref.dataset_id,
                table_id=table_ref.table_id,
                max_results=self.batch_size,
                selected_fields=self.selected_fields,
                start_index=i * self.batch_size,
            )
            rows = [dict(r) for r in response]
            if len(rows) == 0:
                self.log.info('Job Finished')
                return

            self.log.info('Total Extracted rows: %s', len(rows) + i * self.batch_size)

            table_data = []
            for dict_row in rows:
                single_row = []
                for fields in dict_row:
                    single_row.append(dict_row[fields])
                table_data.append(single_row)

            self.log.info(table_data)

            yield table_data
            i += 1
```

**Reply by dudutch** [reply]
I'm attaching my entire bigquery_to_mssql.py that I needed to update.
[bigquery_to_mssql.py.txt](https://github.com/apache/airflow/files/7368784/bigquery_to_mssql.py.txt)

**Reply by josh-fell** [reply]
Agreed, looks related. You could definitely contribute a PR though! Seems like you'd be a great person to fix and test the change.
